src/declarations/stencil-public-compiler.ts

```diff
@@ -2192,30 +2192,268 @@ export type OutputTarget =
   | OutputTargetStats
   | OutputTargetDistTypes;
 
+/**
+ * Our custom configuration interface for generated caching Service Workers
+ * using the Workbox library (see https://developer.chrome.com/docs/workbox/).
+ *
+ * Although we are using Workbox we are unfortunately unable to depend on the
+ * published types for the library because they must be compiled using the
+ * `webworker` lib for TypeScript, which cannot be used at the same time as
+ * the `dom` lib. As a workaround the properties are copy-pasted here from
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c7b4dadae5b320ad1311a8f82242b8f2f41b7b8c/types/workbox-build/generate-sw.d.ts#L3
+ */
 export interface ServiceWorkerConfig {
-  // https://developers.google.com/web/tools/workbox/modules/workbox-build#full_generatesw_config
+  swSrc?: string;
   unregister?: boolean;
 
+  /**
+   * The local directory you wish to match `globPatterns` against. The path is relative to the current directory.
+   */
+  globDirectory?: string;
+
+  /**
+   * The path and filename of the service worker file that will be created by the
+   * build process, relative to the current working directory. It must end in '.js'.
+   */
   swDest?: string;
-  swSrc?: string;
-  globPatterns?: string[];
-  globDirectory?: string | string[];
+
+  /**
+   * A list of entries to be precached, in addition to any entries that are
+   * generated as part of the build configuration.
+   */
+  additionalManifestEntries?: any[] | undefined;
```

Comment:
I don't see this in the diff prior to this commit, nor mention of adding new fields in the PR summary. 

In [this comment](https://github.com/ionic-team/stencil/pull/3813#discussion_r1019400967), you mentioned:

> as I note here we can't at present depend on these types because you need to have the `webworker` lib enabled (and that conflicts with the `dom` lib so both can't be set at the same time 😢).

> If we have project references enabled at some time in the future, allowing parts of the codebase to be compiled as separate units, I think we'd be able to carve off the service-worker related code into it's own project and compile it against that lib instead, but for now we'll have to copy-paste this :/

But I think I'm missing a piece of the puzzle here. What necessitates updating these types? Do they come from the latest version of Workbox? If not, where does this field and these new fields come from?

- babelPresetEnvTargets
- cleanupOutdatedCaches
- globFollow
- globStrict
- importScripts
- inlineWorkboxRuntime
- mode
- navigationPreload
- offlineGoogleAnalytics
- sourcemap

Reply:
yeah the type is just copy-pasted out of workbox. Unfortunately we can't depend on their type declarations directly. Since they are dealing with service worker stuff they depend on having the service worker lib enabled in TypeScript and they therefore won't compiled without our current project setup, since you can't have the 'dom' and service worker libs enabled at the same time (they fight over the typings for a few global names). So I figured the best course is to at least have a more updated interface!

I'll go through and just keep whatever we were 'custom typing' things as before when there's a conflict.

Reply:
Ok! So I looked at this a bit again and decided to just back out this part of the change.

For a little context, what I tried to do was change our setup to use the types for workbox which are published on DefinitelyTyped, I wasn't able to do that because they depend on the `webworker` lib being set and so they won't compile in our project, _but_ in trying to do that I noticed that our config was different than what's published, so I thought we'd want to pull that interface in somehow. I think for now it will be easier to just keep using this interface we maintain though, and we can revisit sometime in the future.